docs/developer/develop-minikube.md

```diff
@@ -0,0 +1,178 @@
+# Develop local with VSCode and Minikube
+
+## Introduction
+This document will describe how to setup a local development environment using VSCode as the IDE,  Minikube as the k8s cluster to run Turbinia and Skaffold to handle the development cycle. 
+
+This setup will provide run time debugging with breakpoints/watches in VSCode as well as hot-reloading of code changes to the live Turbinia deployment without having to rebuild the containers or restart the deployment.
+
+---
+NOTE: This setup has been tested and is in active use by the Turbinia developers in the following configurations.
+* MacAir M3 8GB with Docker Desktop
+* GCE (e2-standard-4) Debian 12/bookworm 16GB with Docker Engine (Running on GCP using the [VSCode Remote SSH extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh))
+   * Setting up the VSCode Remote-SSH extension is outside of scope for this documentation but more information can be found [here](https://code.visualstudio.com/docs/remote/ssh).
+
+## Components
+### Minikube
+As the main supported way to run Turbinia is on k8s it makes sense to also develop as close to a production setup as possible. [Minikube](https://minikube.sigs.k8s.io/docs/) is local Kubernetes cluster so you can run the complete Turbinia stack from your laptop.
+
+### Skaffold
+[Skaffold](https://skaffold.dev/) is a development workflow manager that takes care of building and deploying and monitoring for code changes so you can focus on writing code instead of manually building or restarting containers after every code change.
+
+### VSCode + Google Cloud Code
+VSCode is an IDE that combined with the Google Cloud Code extension will manage the dependencies needed for the development workflow and provide an interface to debug through breakpoints and watches.
+
+---
+## Setup
+### Requirements
+Please install the following requirements on your system.
+* [Docker](https://docs.docker.com/engine/install/), you can install Docker Desktop or only the engine.
+   * Make sure docker can be execute as non-root (`docker run hello-world`). See [here](https://docs.docker.com/engine/install/linux-postinstall/) for instructions.
+   * Some operating systems can run Minikube (faster) with other hypervisors. If you choose so expect some extra custom setup to be done. We advice the use of Docker for Minikube.
+* [Helm](https://helm.sh/docs/intro/install/), to manage the deployment of Turbinia.
+* [VSCode](https://code.visualstudio.com/Download), our IDE of choice for this setup.
+
+### Configure VSCode
+Start VSCode and install the following extensions:
+* [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+* [Gemini Code Assist + Google Cloud Code](https://marketplace.visualstudio.com/items?itemName=GoogleCloudTools.cloudcode)
```

Comment:
So is this only needed when doing development in GCE then?  I can't seem to load the link here right now so not sure how these pieces fit together, but is the Gemini code assist part required?

Reply:
No, the extension will manage *all* the dependencies (among others: minikube, kubectl and skaffold) for you independent if your machine is a local or a remote GCE. It will install and configure the dependecies in a way that do not interfere with your machine configuration. 

The Gemini code assist is included as of lately (unfortunatly..?) and is not activated by default.

Reply:
Interesting, so the only thing we're using it for is dependency management and don't actually use for anything cloud?  What form are all those dependencies in, are they just vscode extensions or are there other packages it installs too?  Is there maybe a lighter weight or better way to do that, or is that just easiest?

Could we maybe add a note clarifying exactly what it's used for (and not used for, e.g. actual cloud management) and that Gemini is not used or enabled by default?

Reply:
The exension manages minikube, skaffold, kubectl and their setup and configuration with a single Install click of the extension. I can't think of an easier way. It will also install these dependencies in a seperate folder so it will not interfere with your system. 

The extension *will* be using the Cloud component for the GKE development option to be added in a further PR.

Gemini code assist is not enabled by default, added a note.

The extension and its use is described in more details at the top of this document in the Components section.